gdnative-core/src/nativescript/init/property/hint.rs

```diff
@@ -116,30 +119,31 @@ where
 /// ```
 #[derive(Clone, Eq, PartialEq, Debug, Default)]
 pub struct EnumHint {
-    values: Vec<String>,
+    values: HashMap<String, i32>,
+}
+
+impl From<HashMap<String, i32>> for EnumHint {
+    #[inline]
+    fn from(values: HashMap<String, i32>) -> Self {
+        EnumHint { values }
+    }
 }
 
 impl EnumHint {
     #[inline]
     pub fn new(values: Vec<String>) -> Self {
-        EnumHint { values }
+        let values: HashMap<_, _> = values
+            .into_iter()
+            .enumerate()
+            .map(|(a, b)| (b, a as i32))
+            .collect();
+        EnumHint::from(values)
     }
 
     /// Formats the hint as a Godot hint string.
     fn to_godot_hint_string(&self) -> GodotString {
-        let mut s = String::new();
-
-        let mut iter = self.values.iter();
-
-        if let Some(first) = iter.next() {
-            write!(s, "{}", first).unwrap();
-        }
-
-        for rest in iter {
-            write!(s, ",{}", rest).unwrap();
-        }
-
-        s.into()
+        let keys: Vec<String> = self.values.keys().cloned().collect();
+        keys.join(",").into()
```

Comment:
It looks like you're not using the values (i.e. associated numbers), are you?

Great how you reduced the code to be much more idiomatic 🙂

Reply:
> It looks like you're not using the values (i.e. associated numbers), are you?

I didn't aware of that before, so I read the code about export a property in gdnative, and found this line at  [`godot-headers`](https://github.com/godotengine/godot-headers), it seems that I can't export information more than enum names?  If so, maybe I should find a workaround to export enum name and its corresponding value?

https://github.com/godotengine/godot-headers/blob/4bae481fc305639a076a69485bed37679149bd32/nativescript/godot_nativescript.h#L56

Reply:
I'll reply to the whole issue, as this affects how we continue.